src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/PaymentsTest.java

```diff
@@ -0,0 +1,148 @@
+package uk.gov.hmcts.reform.bulkscan.endtoendtests;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.Await;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.Container;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.StorageHelper;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.helper.ZipFileHelper;
+import uk.gov.hmcts.reform.bulkscan.endtoendtests.utils.ProcessorEnvelopeResult;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.entity.ContentType.APPLICATION_FORM_URLENCODED;
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.bulkscan.endtoendtests.utils.ProcessorEnvelopeStatusChecker.getZipFileStatus;
+
+public class PaymentsTest {
+
+
+    private static final String REDIRECT_URI = "http://localhost/receiver";
+    public static final String BEARER = "Bearer ";
+
+
+    private Config conf = ConfigFactory.load();
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Test
+    public void should_upload_blob_and_create_exception_record() throws Exception {
+
+        var zipArchive = ZipFileHelper.createZipArchive("test-data/new-application-payments", Container.BULKSCAN);
+
+        StorageHelper.uploadZipFile(Container.BULKSCAN, zipArchive);
+
+        Await.envelopeDispatched(zipArchive.fileName);
+        Await.envelopeCompleted(zipArchive.fileName);
+
+        //get the process result again and assert
+        String ccdId = assertCompletedProcessorResultAndRetrieveCcdId(zipArchive.fileName);
+
+        String accessToken = getIdamToken();
+
+        String s2sToken = getS2SToken();
```

Comment:
do we need to define those variables we could have directly used `getIdamToken()` and `getS2SToken()` below

Reply:
I prefer this style. If you insist I can change

Reply:
yes please